### DISCUSSION TITLE
LTE and 5G Network Connection Failure: POSIXErrorCode(rawValue: 50): Network is down

### DISCUSSION THREAD

**Discussion by DevPlayNew** [HaishinKit/HaishinKit.swift#1457] [Q&A] [answered]
### Describe the bug

Hello,

First, thank you for providing an excellent library. I am currently implementing RTMP streaming using HaishinKit.swift. The implementation works perfectly in a WiFi environment. However, when switching to LTE and 5G environments, the connection fails with the following error log:

====================================
POSIXErrorCode(rawValue: 50): Network is down
====================================

WiFi Environment: The RTMP server connects successfully and streaming works smoothly.
LTE/5G Environment: The connection to the RTMP server fails, resulting in a NetConnection.Connect.Failed error along with POSIXErrorCode(rawValue: 50): Network is down.

Environment Details
HaishinKit.swift Version: Latest version (1.8.1)
iOS Version: iOS 14 and above
Network Environment: LTE and 5G

Troubleshooting Steps Taken
Server Address Verification: Ensured that the RTMP server address is correct. It connects successfully over WiFi.
Network Settings Verification: Verified that the ports are not blocked on LTE and 5G networks.
RTMP Server Configuration: Confirmed that the RTMP server is properly set up and accessible externally.
Log Inspection: Enabled and reviewed HaishinKit logs to diagnose the error cause.
**Example Code Testing**: Tested using the example code provided in the HaishinKit.swift repository to ensure correct implementation.

Inquiry
Could you please provide insights into why the POSIXErrorCode(rawValue: 50): Network is down error occurs in LTE and 5G environments? Additionally, I would like to know if there are any further steps or configurations I should check to resolve this issue. If there is any known bug or configuration related to this issue, your guidance would be greatly appreciated.

Thank you.

### Steps to Reproduce

1. **Set up an RTMP server** that is accessible from both WiFi and cellular networks.
2. **Configure the HaishinKit.swift example app** to connect to the RTMP server.
3. **Connect the device to a WiFi network** and start streaming. Verify that the connection is successful.
4. **Switch the device to an LTE or 5G network**.
5. Attempt to start streaming again.

### Expected behavior

The app should connect to the RTMP server and start streaming successfully over LTE and 5G networks, just as it does over WiFi.

### Version

1.8.1

**Comment by shogo4405** [comment]
The error is coming from the [Network.framework](https://developer.apple.com/documentation/network) that I'm using internally. Honestly, I don't have much knowledge about it. Regarding the error related to Network.framework, I believe you can find tutorials by looking it up.

For instance, do you experience the same issue with external sites like YouTube or Facebook?
There have been instances where the connection was blocked by the network provider.
In such cases, using `rtmps` resolved the issue. https://github.com/shogo4405/HaishinKit.swift/issues/83

**Comment by DevPlayNew** [comment]
Hello,

Thank you for your response.

As you suggested, I checked whether the same issue occurs with external sites like YouTube and Facebook. The streaming works correctly on these external sites.

Additionally, I built and tested the streaming using HaishinKit version 1.2.2 on LTE and 5G networks, and it worked successfully. Therefore, it seems that the issue only occurs with the latest version (1.8.1).

Based on this information, could you please provide further assistance? If there are any specific settings or solutions to this issue, I would appreciate your guidance.

Thank you.

**Comment by shogo4405** [comment]
In that version, NSInputStream/NSOutputStream is used, and it may be possible to connect. On the other hand, there is a higher tendency to crash and it takes longer to reconnect when switching networks.

Although it was removed in version 1.8.0, in the 1.7.x series, it is possible to fall back to NSInputStream/NSOutputStream by setting `RTMPConnection.requireNetworkFramework = false`.

It is highly likely that the compatibility between Network.framework and the network infrastructure being used is causing the issue, so I recommend experimenting with the options provided by Network.framework.

**Reply by DevPlayNew** [reply, marked as answer]
Thank you, your help resolved the issue.

RTMPConnection.requireNetworkFramework = false.

**Comment by shogo4405** [comment]
It seems like it might be a configuration issue with Network.framework, so I plan to move this to the Discussion section to gather insights.
